lib/resources/df.rb

```diff
@@ -14,15 +12,26 @@ class DfResource < Inspec.resource(1)
 
     def initialize(partition)
       @partition = partition
-      @cmd = inspec.command("df #{@partition} --output=size | sed \"/blocks/d; s/ *//g\"")
     end
 
     def size
-      @cmd.stdout.to_i
+      if inspec.os.linux?
+        command = "df #{partition} --output=size"
+      else
+        raise Inspec::Exceptions::ResourceFailed, 'The `df` resource is not supported on your OS.'
```

Comment:
The `supports` stuff you added takes care of this for you so you can remove all of this logic and just declare your command. 🙂 

And you should move the command into the `@size` memoifier so it looks something like this and is the first line of the `size` method:

```ruby
@size ||= begin
  cmd = inspec.command("df #{partition} --output=size")
  raise ...
```

Reply:
I thought as much as well, but would it be nice to have for future support of other OS families?

Reply:
Sure, but if we're not going to account for that in this PR, we should leave it out.

If we add future family support, I'd recommend doing two things:

1. Flag the resource for the additional family support using the `supports` method
2. Change the line to:

```ruby
cmd = inspec.command(df_command_for_family)
```

... and supply a new `df_command_for_family` method that returns a command string appropriate for the node under test.